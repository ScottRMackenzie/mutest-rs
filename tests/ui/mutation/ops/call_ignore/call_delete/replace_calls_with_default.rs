//@ print-mutations
//@ build
//@ stdout
//@ stderr: empty
//@ mutation-operators: call_delete

#![allow(unused)]

struct S<T> {
    v: usize,
    t: T,
}

impl<T: Default> Default for S<T> {
    fn default() -> Self {
        Self { v: 0, t: Default::default() }
    }
}

fn f() {
    fn g(_: ()) -> S<usize> {
        S { v: 0, t: 1 }
    }

    let _ = g(());

    fn h<T>(t: T) -> S<T> {
        S { v: 0, t }
    }

    let _ = h(1_i32);
    let _ = h(&mut false);
}

#[test]
fn test() {
    f();
}
