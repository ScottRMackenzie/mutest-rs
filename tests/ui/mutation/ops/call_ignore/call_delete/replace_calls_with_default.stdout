[call_delete] delete call to `f::g` and replace it with `Default::default()` in f at tests/ui/mutation/ops/call_ignore/call_delete/replace_calls_with_default.rs:25:13: 25:18
  <-(0)- test

[call_delete] delete call to `f::h` and replace it with `Default::default()` in f at tests/ui/mutation/ops/call_ignore/call_delete/replace_calls_with_default.rs:31:13: 31:21
  <-(0)- test

2 mutations; 2 safe; 0 unsafe (0 tainted)
